src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/AppleCertificatePal.cs

```diff
@@ -667,7 +667,26 @@ private void EnsureCertData()
                 return;
 
             Debug.Assert(!_certHandle.IsInvalid);
-            _certData = new CertificateData(Interop.AppleCrypto.X509GetRawData(_certHandle));
+            string? subjectSummary = Interop.AppleCrypto.X509GetSubjectSummary(_certHandle);
```

Comment:
Why can't we do this only in the case of failure?  This is adding cost that appears not to be used in the success case.

Reply:
Good point.  We should fix that in main.

Reply:
We should fix that in both :)